### PR TITLE
[Onboarding] handle Facebook login errors gracefully

### PR DESCRIPTION
**Pull request by l2succes**
Upgrades Facebook iOS SDK to `4.33.0` and allows to us to handle graph request errors

### CODE REVIEW

File: Artsy/Networking/ARAuthProviders.m [hunk 1]
```diff
@@ -34,12 +34,18 @@ + (void)getTokenForFacebook:(void (^)(NSString *token, NSString *email, NSString
 
         } else {
             FBSDKGraphRequest *request = [[FBSDKGraphRequest alloc] initWithGraphPath:@"me?fields=name,id,email" parameters:nil];
+            
+            // The next line is important.
+            // It prevents FB SDK from handling the error automatically
+            // and allows you to handle the error yourself.
+            // In this way you avoid(!) the FBSDK dialog that says
+            // "Please login to this app again to reconnect with Facebook".
+            [request setGraphErrorRecoveryDisabled:YES];
```
Comment (orta) [inline Artsy/Networking/ARAuthProviders.m]: I think we could improve this comment to be a bit more on-brand

```
// We need to disable FB's in-house error reporting so we can show our own
```

File: Artsy/App/ARAppDelegate.m [hunk 2]
```diff
@@ -145,7 +145,6 @@ - (void)setupForAppLaunch
     _landingURLRepresentation = self.landingURLRepresentation ?: @"https://artsy.net";
 
     [[ARLogger sharedLogger] startLogging];
-    [FBSDKSettings setAppID:[ArtsyKeys new].artsyFacebookAppID];
```
[the remark this answers is not in the document]
Comment (l2succes) [reply Artsy/App/ARAppDelegate.m]: Yes, it reads the value from the app's `Info.plist` if you don't set it.

From the docs:
screen shot 2018-05-15 at 3 15 54 pm

[the remark this answers is not in the document]
Comment (l2succes) [reply Artsy/App/ARAppDelegate.m]: That being said, the reason I removed was to override the value in testing so I'm fine with putting it back in. It's just not needed

**Comment by l2succes** [comment]
https://circleci.com/gh/artsy/eigen/5221

**Pull request merged**